Gems/PhysX/Code/Source/EditorMeshColliderComponent.cpp

```diff
@@ -0,0 +1,1101 @@
+    {
+        return m_shapeConfiguration;
+    }
+
+    const Physics::ColliderConfiguration& EditorMeshColliderComponent::GetColliderConfiguration() const
+    {
+        return m_configuration;
+    }
+
+    Physics::ColliderConfiguration EditorMeshColliderComponent::GetColliderConfigurationScaled() const
+    {
+        // Scale the collider offset
+        Physics::ColliderConfiguration colliderConfiguration = m_configuration;
+        colliderConfiguration.m_position *= Utils::GetTransformScale(GetEntityId()) * m_cachedNonUniformScale;
+        return colliderConfiguration;
+    }
+
+    Physics::ColliderConfiguration EditorMeshColliderComponent::GetColliderConfigurationNoOffset() const
+    {
+        Physics::ColliderConfiguration colliderConfiguration = m_configuration;
+        colliderConfiguration.m_position = AZ::Vector3::CreateZero();
+        colliderConfiguration.m_rotation = AZ::Quaternion::CreateIdentity();
+        return colliderConfiguration;
+    }
+
+    void EditorMeshColliderComponent::Activate()
+    {
+        m_sceneInterface = AZ::Interface<AzPhysics::SceneInterface>::Get();
+        if (m_sceneInterface)
+        {
+            m_editorSceneHandle = m_sceneInterface->GetSceneHandle(AzPhysics::EditorPhysicsSceneName);
+        }
+
+        m_physXConfigChangedHandler = AzPhysics::SystemEvents::OnConfigurationChangedEvent::Handler(
+            []([[maybe_unused]] const AzPhysics::SystemConfiguration* config)
+            {
+                AzToolsFramework::PropertyEditorGUIMessages::Bus::Broadcast(&AzToolsFramework::PropertyEditorGUIMessages::RequestRefresh,
+                    AzToolsFramework::PropertyModificationRefreshLevel::Refresh_AttributesAndValues);
+            });
+
+        AzToolsFramework::Components::EditorComponentBase::Activate();
+        AzToolsFramework::EntitySelectionEvents::Bus::Handler::BusConnect(GetEntityId());
+        PhysX::MeshColliderComponentRequestsBus::Handler::BusConnect(GetEntityId());
+        AZ::TransformNotificationBus::Handler::BusConnect(GetEntityId());
+        AzToolsFramework::ShapeManipulatorRequestBus::Handler::BusConnect(
+            AZ::EntityComponentIdPair(GetEntityId(), GetId()));
+        ColliderShapeRequestBus::Handler::BusConnect(GetEntityId());
+        AZ::Render::MeshComponentNotificationBus::Handler::BusConnect(GetEntityId());
+        EditorColliderComponentRequestBus::Handler::BusConnect(AZ::EntityComponentIdPair(GetEntityId(), GetId()));
+        EditorMeshColliderComponentRequestBus::Handler::BusConnect(AZ::EntityComponentIdPair(GetEntityId(), GetId()));
+        EditorMeshColliderValidationRequestBus::Handler::BusConnect(GetEntityId());
+        AzFramework::BoundsRequestBus::Handler::BusConnect(GetEntityId());
+        AzToolsFramework::EditorComponentSelectionRequestsBus::Handler::BusConnect(GetEntityId());
+        m_nonUniformScaleChangedHandler = AZ::NonUniformScaleChangedEvent::Handler(
```

Comment:
@greerdv Maybe non-uniform scale stuff can now be removed for this type? Or is it still needed?

Reply:
No, you can have non-uniform scale with asset colliders. In fact, that was the reason we made collider component compatible with NUS in the first place. Since prior to this change, the collider component supported both assets and primitives, it actually meant a bunch of work had to be done to replace primitives with convex meshes if they had NUS. If this change had happened first, we could actually have just made primitive colliders incompatible with NUS component.

Reply:
As an aside, I would really like to either get rid of asset scale or make it uniform. I don't think having NUS in there adds value, since if you want it you can use the NUS component. But it does break primitives inside asset colliders when there is no NUS component.

Reply:
> As an aside, I would really like to either get rid of asset scale or make it uniform. I don't think having NUS in there adds value, since if you want it you can use the NUS component. But it does break primitives inside asset colliders when there is no NUS component.

+1 on this as a follow up task

Reply:
Asset scale is just for the collider mesh/convex while NUS component applies for the entire entity. 
There may be use cases when you want to adjust the scale just for the collision geom without changing the entity scale.